src/snowflake/snowpark/relational_grouped_dataframe.py

```diff
@@ -666,13 +670,16 @@ def function(self, agg_name: str, _emit_ast: bool = True) -> Callable:
 
     builtin = function
 
+    @publicapi
     def _function(
```

Comment:
why is this a publicapi?

Reply:
I mainly added the decorator as a safeguard that checks the global AST enabled flag before trying to generate the AST by default, rather than to signal that this method can be used publicly.